src/components/Checkout/CheckoutAPICalls.tsx

```diff
@@ -33,20 +33,22 @@ export async function processGeneralItems(user: ClientPrincipal | null, loggedIn
   return await response.json();
 }
 
-export async function getUnitNumbers(buildingId: number) {
+export async function getUnitNumbers(user: ClientPrincipal | null, buildingId: number) {
+  API_HEADERS['X-MS-API-ROLE'] = getRole(user);
   try {
-    const response = await fetch(`${ENDPOINTS.UNITS}?$filter=(building_id eq ${buildingId})`, {
+    const response = await fetch(`${ENDPOINTS.UNITS}?$filter=building_id eq ${buildingId}&$first=200`, {
```

Comment:
Let's increase the margin in case a building with 250 units is added. 

```suggestion
    // By default, API fetches only 100 items. Some buildings have more than 100 units. 
    // To be safe, let's fetch up to 1000
    const response = await fetch(`${ENDPOINTS.UNITS}?$filter=building_id eq ${buildingId}&$first=1000`, {
```